sr_utilities_common/scripts/sr_utilities_common/conditional_delayed_rostool.py

```diff
@@ -25,16 +25,29 @@ class RosElementsHandler(object):
     def __init__(self, element_type, required_elements_list):
         self._element_type = element_type
         self._required_elements_list = required_elements_list
+        self._found_elements = []
         self.missing_elements = []
 
-    def execute(self):
-        all_elements_available = False
+    def check_if_required_element_is_available(self):
         roscore_published_elements = self._retrieve_available_elements(self._element_type)
-        self._check_if_element_is_available(roscore_published_elements, self._required_elements_list)
-        if 0 == len(self._required_elements_list):
-            all_elements_available = True
-        self.missing_elements = self._required_elements_list
-        return all_elements_available
+        if isinstance(self._required_elements_list, list):
```

Comment:
please do
if not is instance:
    raise
and just continue with code. makes it more readable, no need to add else below or wrap more code up in an if